modules/bindbackend/binddnssec.cc

```diff
@@ -323,8 +323,8 @@ bool Bind2Backend::getDomainKeys(const DNSName& name, std::vector<KeyData>& keys
     SSqlStatement::row_t row;
     while (d_getDomainKeysQuery_stmt->hasNextRow()) {
       d_getDomainKeysQuery_stmt->nextRow(row);
-      kd.id = pdns_stou(row[0]);
-      kd.flags = pdns_stou(row[1]);
+      kd.id = pdns::checked_stou<decltype(kd.id)>(row[0]);
```

Comment:
I think we can merge `pdns::checked_stou` and `pdns::checked_stoi` based on the type, if we want to:
```c++
template <typename T>
auto checked(const std::string& str, size_t* idx = nullptr, int base = 10) -> T
{
  static_assert(std::numeric_limits<T>::is_integer, "checked: The `T` type must be an integer");

  if (str.empty()) {
    return 0; // compatibility                                                                                                                                                        
  }

  if constexpr (std::is_unsigned_v<T>) {
    return pdns::checked_conv<T>(std::stoull(str, idx, base));
  }
  else {
    return pdns::checked_conv<T>(std::stoll(str, idx, base));
  }
}
```

Did you consider writing a version of `pdns::checked_*` taking the destination as a parameter instead of returning it, so that we can avoid the duplication of the target name? Something like: `pdns::checked_stou_into(row[0], kd.id);`.
A macro would likely also work:
```c++
#define pdns_checked_stoi_into(str, dest) \
  dest = pdns::checked_stoi<decltype(dest)>(str)

#define pdns_checked_stou_into(str, dest) \
  dest = pdns::checked_stou<decltype(dest)>(str)
```

Reply:
I'm all for merging `pdns::checked_stou` into `pdns::checked_stoi` with the use of `constexpr` and already made the change. Thank you!

As for `checked_stoi_into`, that's also fine (I prefer it to be a function over a macro). However, I must mention that its usage breaks the flow of readability - it's very sad that the compiler is unable to infer the target type of `checked_stoi`. Example:

```c++
r.qname = DNSName(parts[1 + extraFields]);
r.qtype = parts[3 + extraFields];
pdns::checked_stoi_into(parts[4 + extraFields], r.ttl);
pdns::checked_stoi_into(parts[5 + extraFields], r.domain_id);
```

Reply:
Note: The first parameter of `checked_stoi_into` is the out parameter (the target).